File: src/JavaScript/Tests/TestRunner.js
class TestRunner {
  static runTests(challenge, inputs, expectedResults) {
    let failed = 0;

    for (let i = 0; i < inputs.length; i++) {
      try {
        const actual = challenge.f(inputs[i]);
        if (actual !== expectedResults[i]) {
          console.log(
            `❌ Test ${i + 1} failed: input=${inputs[i]} expected="${expectedResults[i]}" got="${actual}"`
          );
          failed++;
        }
      } catch (e) {
        console.log(`💥 Test ${i + 1} errored: ${e.message}`);
        failed++;
      }
    }

    return failed;
  }
}

module.exports = TestRunner;

File: src/Java/Tests/TestRunner.java
package Java.Tests;

public class TestRunner {
    public static <I, O> int runTests(Challenge<I, O> challenge, I[] inputs, O[] expected) {
        int failed = 0;
        for (int i = 0; i < inputs.length; i++) {
            try {
                O result = challenge.f(inputs[i]);
                if (!result.equals(expected[i])) {
                    System.out.printf("❌ Test #%2d FAILED — Input: %s Expected: %s Got: %s%n",
                                      i + 1, inputs[i], expected[i], result);
                    failed++;
                }
            } catch (Exception e) {
                System.out.printf("Test #%2d ERROR — Input: %s threw %s%n", i + 1, inputs[i], e);
                failed++;
            }
        }
        return failed;
    }


}